Improve developer docs

mprpic opened the pull request.

mprahl commented on README.md:
````diff
@@ -5,8 +5,16 @@ A system for discovering and tracing the composition of shipped products.
+* Create and activate a [Python virtual environment](https://docs.python.org/3/library/venv.html)
+    (Python 3.6 or later is required):
+  ```bash
+  $ python3 -m venv --system-site-packages venv
````
I don't think we want to use `--system-site-packages` since it can lead to "works on my machine" issues.

mprpic replied:
How else do you get the `rpm` module into your venv then?

mprahl replied:
Does `pip install rpm` provide a different dependency than you were expecting?

mprpic replied:
That package is just a stub on [PyPI](https://pypi.org/project/rpm/):

```python
>>> import rpm
/home/.../venv/lib64/python3.6/site-packages/rpm.py:15: UserWarning: The RPM Python bindings are not currently available via PyPI.

Please install them with your distro package manager (typically called
'python2-rpm' or 'python3-rpm'), and ensure that any virtual environments
needing the API are configured to be able to see the system site packages
directory.
```

mprahl replied:
Okay, you should mention that you need to install the RPM then before creating the virtualenv.

mprpic replied:
It's installed by default on all RPM-based systems (dnf/yum require it as a dependency). The point here is that if you create an empty venv, it won't be available unless you allow that venv access to the system packages. I don't think it's possible to specify that you only want to pull in the `rpm` module but not anything else from the system unless you do some manual symlinking.

mprahl commented on README.md:
````diff
@@ -5,8 +5,16 @@ A system for discovering and tracing the composition of shipped products.
+  ```bash
+  $ python3 -m venv --system-site-packages venv
+  ```
+* [Install Neo4j](https://neo4j.com/docs/operations-manual/current/installation/linux/rpm/) and
````
This is fine, but should we encourage the use of containers instead? For the community edition, it's as simple as:
```bash
$ sudo docker run --publish=7474:7474 --publish=7687:7687 neo4j:3.3
```

mprahl commented:
@mprpic could you please update your commits to use your GitHub email as the author?

mprpic commented:
> @mprpic could you please update your commits to use your GitHub email as the author?

Done!

mprpic committed:
Ignore PyCharm dir; add missing command prompt

mprpic committed:
Add neo4j install instructions, clarify Python venv

koji requires the python3-rpm dependency, which cannot be installed from
PyPI. It's installed by default on all RPM-based systems so when running
from a virtual env, include it from the system packages.

mprahl commented:
:+1:
